src/Integration.Vsix/SonarLintDaemon/CFamily.cs

```diff
@@ -0,0 +1,471 @@
+                Add(c.Cmd, "true".Equals(OpenMPSupport) ? "/openmp" : "");
+
+                Add(c.Cmd, ConvertRuntimeLibrary(RuntimeLibrary));
+                Add(c.Cmd, ConvertExceptionHandling(ExceptionHandling));
+                Add(c.Cmd, ConvertEnableEnhancedInstructionSet(EnableEnhancedInstructionSet));
+                Add(c.Cmd, "true".Equals(OmitDefaultLibName) ? "/Zl" : ""); // defines macro "_VC_NODEFAULTLIB"
+                Add(c.Cmd, "false".Equals(RuntimeTypeInfo) ? "/GR-" : ""); // undefines macro "_CPPRTTI"
+                Add(c.Cmd, ConvertBasicRuntimeChecks(BasicRuntimeChecks));
+
+                // TODO Q: what if it contains space in double quotes?
+                AddList(c.Cmd, AdditionalOptions.Split(' '));
+
+                c.Cmd.Add(AbsoluteFilePath);
+
+                return new Capture[] { p, c };
+            }
+
+            public string ToJson(string path, out string sqLanguage)
+            {
+                Capture[] c = ToCaptures(path, out sqLanguage);
+                string json = "{ \"version\": 0, \"captures\": [ " + c[0].ToJson() + " , " + c[1].ToJson() + " ] }";
+                System.Diagnostics.Debug.WriteLine(json);
+                return json;
+            }
+
+            internal static string ConvertPlatformName(string platformName)
+            {
+                switch (platformName)
+                {
+                    default:
+                        throw new Exception("Unsupported PlatformName: " + platformName);
+                    case "Win32":
+                        return "x86";
+                    case "x64":
+                        return "x64";
+                }
+            }
+
+            internal static string ConvertPlatformToolset(string platformToolset)
+            {
+                switch (platformToolset)
+                {
+                    default:
+                        throw new Exception("Unsupported PlatformToolset: " + platformToolset);
+                    case "v141":
+                    case "v141_xp":
+                        return "19.10.00";
+                    case "v140":
+                    case "v140_xp":
+                        return "19.00.00";
+                    case "v120":
+                    case "v120_xp":
+                        return "18.00.00";
+                    case "v110":
+                    case "v110_xp":
+                        return "17.00.00";
+                    case "v100":
+                        return "16.00.00";
+                    case "v90":
+                        return "15.00.00";
+                }
+            }
+
+            internal static string ConvertCompileAsAndGetSqLanguage(string value, string path, out string sqLanguage)
+            {
+                switch (value)
+                {
+                    default:
+                        throw new Exception("Unsupported CompileAs: " + value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'System.Exception' should not be thrown by user code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS112)